src/vs/workbench/contrib/chat/common/codeBlockModelCollection.ts

```diff
@@ -11,15 +11,16 @@ import { Range } from '../../../../editor/common/core/range.js';
 import { ILanguageService } from '../../../../editor/common/languages/language.js';
 import { EndOfLinePreference } from '../../../../editor/common/model.js';
 import { IResolvedTextEditorModel, ITextModelService } from '../../../../editor/common/services/resolverService.js';
+import { extractCodeblockUrisFromText, extractVulnerabilitiesFromText, IMarkdownVulnerability } from './annotations.js';
 import { IChatRequestViewModel, IChatResponseViewModel, isResponseVM } from './chatViewModel.js';
-import { extractVulnerabilitiesFromText, IMarkdownVulnerability } from './annotations.js';
 
 
 export class CodeBlockModelCollection extends Disposable {
 
 	private readonly _models = new ResourceMap<{
 		readonly model: Promise<IReference<IResolvedTextEditorModel>>;
 		vulns: readonly IMarkdownVulnerability[];
+		codemapperUri?: URI;
```

Comment:
I wonder if you'd just want to call it `uri` internally, it could be used for anything

Reply:
Seems dangerous, there are places where we have `uri` already for a codeblock https://github.com/microsoft/vscode/blob/0c0f1d823be7affad7380740a38c63ad59a55d14/src/vs/workbench/contrib/chat/browser/chat.ts#L82-L83

Reply:
Oh yeah, the codeblock text model does have an actual uri, so best to not confuse the two...